.circleci/config.yml

```diff
@@ -2,53 +2,80 @@ version: 2.1
 orbs:
   anchore: anchore/anchore-engine@1.4.0
 jobs:
-  Full Build:
-    machine:
+  Assemble:
+    machine: &virtual_machine
       image: circleci/classic:latest
       docker_layer_caching: true
     steps:
-      - checkout
-      - run:
-          name: Install java 11
+      - run: &install_java
+          name: Install Java 11
           command: |
             sudo add-apt-repository ppa:openjdk-r/ppa
-            sudo apt-get update -q
-            sudo apt install -y openjdk-11-jdk
-      - run:
-          name: Set default java 11
-          command:
-            sudo update-java-alternatives --set java-1.11.0-openjdk-amd64
-      - run:
-          name: Building
-          command: |
-            ./gradlew clean build --info
-      - run:
-          name: Save test results
-          command: |
-            mkdir -p ~/test-results/junit/
-            find . -type f -regex ".*/target/surefire-reports/.*xml" -exec cp {} ~/test-results/junit/ \;
-          when: always
-      - store_test_results:
-          path: ~/test-results
-      - store_artifacts:
-          path: ~/test-results/junit
+            sudo apt-get -qq update
+            sudo apt-get -qq install -y openjdk-11-jdk
+      - run: &configure_java
+          name: Change to Java 11
+          command: sudo update-java-alternatives --set java-1.11.0-openjdk-amd64
+      - checkout
+      - restore_cache: &restore_cache
+          keys:
+            # Use increasingly general patterns to restore cache
+            - gradle-repo-v1-{{ .Branch }}-
+            - gradle-repo-v1-
+      - run: ./gradlew assemble --info
+        # Cache local artifact repository
+      - save_cache:
+          paths:
+            - ~/.gradle
+            - ~/.m2
+          key: gradle-repo-v1-{{ .Branch }}
       - run: mkdir -p workspace
       - run:
           command: 'docker save cnxta/ion-search > workspace/ion-search.tar'
       - persist_to_workspace:
           root: workspace
           paths:
             - ion-search.tar
+  Test:
+    machine: *virtual_machine
+    steps:
+      - run: *install_java
+      - run: *configure_java
+      - checkout
+      - restore_cache: *restore_cache
+      - run: ./gradlew check
```

Comment:
✏️Doesn't the `check` task run the `assemble` task? Looks like it might not make sense to break out the tasks this way.